bauble/src/value/mod.rs

```diff
@@ -459,18 +476,56 @@ pub fn convert_values<C: AssetContext>(
         .into_iter()
         .map(|s| s.to_string())
         .collect::<Vec<_>>();
+    let mut objects = Vec::new();
+    let mut auto_value_idx = 0;
+
+    let mut add_value = |val: Val| {
+        let name = format!("@{auto_value_idx:x}");
+        let span = val.value.span;
+        objects.push(create_object(&path, &name, val));
+        auto_value_idx += 1;
+        Val {
+            value: Value::Ref(format!("{path}::{name}")).spanned(span),
+            attributes: Attributes::default().spanned(span),
+        }
+    };
 
     for item in order {
-        let val = convert_value(&values.copies[item.as_str()], &symbols)?;
+        let val = convert_value(
+            &values.copies[item.as_str()],
+            &symbols,
+            &mut add_value,
+            false,
+        )?;
 
         symbols.uses.insert(item, RefCopy::Resolved(val));
     }
 
-    values
+    let parsed_objects = values
         .values
         .iter()
-        .map(|value| convert_object(&path, &value.0.value, value.1, &symbols))
-        .try_collect()
+        .map(|value| convert_object(&path, &value.0.value, value.1, &symbols, &mut add_value))
+        .collect::<Vec<_>>();
+    {
+        use ariadne::{Color, Label, Report, ReportKind, Source};
+
+        for res in parsed_objects.iter() {
+            if let Err(e) = res {
+                Report::build(ReportKind::Error, (), e.span.start)
+                    .with_message(e.to_string())
+                    .with_label(
+                        Label::new(e.span.into_range())
+                            .with_message("Here")
+                            .with_color(Color::Red),
+                    )
+                    .finish()
+                    .eprint(Source::from(source))
+                    .unwrap()
+            }
+        }
+    }
```

Comment:
Does this need to have its own scope? It doesn't look like it but I don't know `ariadne` well, so I could be missing something.

Reply:
It doesn't need to, but did it more to keep the uses in the scope.